lib/teleterm/daemon/daemon.go

```diff
@@ -813,6 +814,44 @@ func (s *Service) GetKubes(ctx context.Context, req *api.GetKubesRequest) (*clus
 	return response, nil
 }
 
+// ListKubernetesResourcesRequest defines a request to retrieve kube resources paginated.
+// Only one type of kube resource can be retrieved per request (eg: namespace, pods, secrets, etc.)
+func (s *Service) ListKubernetesResources(ctx context.Context, clusterURI uri.ResourceURI, req *api.ListKubernetesResourcesRequest) (*kubeproto.ListKubernetesResourcesResponse, error) {
+	cluster, tc, err := s.ResolveClusterURI(clusterURI)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var proxyGRPCClient kubeproto.KubeServiceClient
+
+	err = clusters.AddMetadataToRetryableError(ctx, func() error {
+		proxyGRPCClient, err = tc.NewKubernetesServiceClient(ctx, cluster.Name)
```

Comment:
Nit: We can put both calls under the same `AddMetadataToRetryableError` call, it results in the same behavior and it's slightly shorter.